assignment/core/src/main/java/com/assignment/core/services/AsymmetricCipherProvider.java

```diff
@@ -0,0 +1,7 @@
+package com.assignment.core.services;
+
+public interface AsymmetricCipherProvider {
+    String getAlgorithm();
+    String encrypt(String text) throws Exception;
```

Comment:
`throws Exception` is too generalised, the exception you are declaring in your interface should be narrower